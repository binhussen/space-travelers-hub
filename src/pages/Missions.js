import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Mission from '../components/Mission';
import { getMissions } from '../redux/missions/missions';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(getMissions());
    }
  }, []);
  return (
    <section className="flex justify-center items-center w-80vw my-5vh mx-auto">
      <table className="border-collapse rounded text-base w-full table-auto border">
        <thead>
          <tr>
            <th className="text-lg text-center border p-3">Mission</th>
            <th className="text-lg text-center border p-3">Description</th>
            <th className="text-lg text-center border p-3" colSpan="2">
              Status
            </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <Mission key={mission.id} mission={mission} />
          ))}
        </tbody>
      </table>
    </section>
  );
};
export default Missions;
