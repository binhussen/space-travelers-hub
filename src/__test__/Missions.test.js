/** @format */

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import missionReducer from '../redux/missions/missions';
import rocketReducer from '../redux/rockets/rockets';
import Missions from '../pages/Missions';

const initialMissionState = {
  missions: [
    {
      id: '9D1B7E0',
      name: 'Thaicom',
      description:
				'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
      joined: false,
    },
  ],
};

const rootReducer = combineReducers({
  missions: missionReducer,
  rockets: rocketReducer,
});

const store = configureStore({
  reducer: rootReducer,
  preloadedState: initialMissionState,
});

describe('Missions tests', () => {
  test('Is Mission Page Render', () => {
    const missions = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(missions).toMatchSnapshot();
  });

  test('Display the Mission page', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    const missionName = screen.getByText('Thaicom');
    expect(missionName).toBeInTheDocument();
  });
});
