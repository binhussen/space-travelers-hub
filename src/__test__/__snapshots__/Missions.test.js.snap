// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions tests Is Mission Page Render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex justify-center items-center w-80vw my-5vh mx-auto"
      >
        <table
          class="border-collapse rounded text-base w-full table-auto border"
        >
          <thead>
            <tr>
              <th
                class="text-lg text-center border p-3"
              >
                Mission
              </th>
              <th
                class="text-lg text-center border p-3"
              >
                Description
              </th>
              <th
                class="text-lg text-center border p-3"
                colspan="2"
              >
                Status
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="border p-3"
              >
                Thaicom
              </td>
              <td
                class="border p-3"
              >
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
              </td>
              <td
                class="border p-3"
              >
                <span
                  class="bg-gray-400 border-0 rounded text-white px-3 py-2 text-base block"
                >
                  Not a member
                </span>
              </td>
              <td
                class="border p-3"
              >
                <button
                  class="bg-sky-500 border-0 rounded text-white cursor-pointer px-3 py-2 text-base"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex justify-center items-center w-80vw my-5vh mx-auto"
    >
      <table
        class="border-collapse rounded text-base w-full table-auto border"
      >
        <thead>
          <tr>
            <th
              class="text-lg text-center border p-3"
            >
              Mission
            </th>
            <th
              class="text-lg text-center border p-3"
            >
              Description
            </th>
            <th
              class="text-lg text-center border p-3"
              colspan="2"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="border p-3"
            >
              Thaicom
            </td>
            <td
              class="border p-3"
            >
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td
              class="border p-3"
            >
              <span
                class="bg-gray-400 border-0 rounded text-white px-3 py-2 text-base block"
              >
                Not a member
              </span>
            </td>
            <td
              class="border p-3"
            >
              <button
                class="bg-sky-500 border-0 rounded text-white cursor-pointer px-3 py-2 text-base"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
